scio-examples/src/main/scala/com/spotify/scio/examples/complete/game/LeaderBoard.scala

```diff
@@ -41,43 +53,62 @@ object LeaderBoard {
 
   // scalastyle:off method.length
   def main(cmdlineArgs: Array[String]): Unit = {
+    // Create `ScioContext` and `Args`
     val (opts, args) = ScioContext.parseArguments[ExampleOptions](cmdlineArgs)
     val sc = ScioContext(opts)
     sc.optionsAs[StreamingOptions].setStreaming(true)
     val exampleUtils = new ExampleUtils(sc.options)
 
+    // Date formatter for full timestamp
     def fmt = DateTimeFormat.forPattern("yyyy-MM-dd HH:mm:ss.SSS")
       .withZone(DateTimeZone.forTimeZone(TimeZone.getTimeZone("PST")))
+    // Duration in minutes over which to calculate team scores, defaults to 1 hour
     val teamWindowDuration = Duration.standardMinutes(args.int("teamWindowDuration", 60))
+    // Data that comes in from our streaming pipeline after this duration isn't considered in our
+    // processing. Measured in minutes, defaults to 2 hours
     val allowedLateness = Duration.standardMinutes(args.int("allowedLateness", 120))
 
+    // Read in streaming data from PubSub and parse each row as `GameActionInfo` events
     val gameEvents = sc.pubsubTopic(args("topic"), timestampAttribute = "timestamp_ms")
       .flatMap(UserScore.parseEvent)
 
     calculateTeamScores(gameEvents, teamWindowDuration, allowedLateness)
+      // Add windowing information to team score results by converting to `WindowedSCollection`
       .toWindowed
       .map { wv =>
+        // Convert from score tuple to TeamScoreSums object with both tuple and windowing info
         val start = fmt.print(wv.window.asInstanceOf[IntervalWindow].start())
         val now = fmt.print(Instant.now())
         val timing = wv.pane.getTiming.toString
         wv.copy(value = TeamScoreSums(wv.value._1, wv.value._2, start, now, timing))
       }
+      // Done with windowing information, convert back to regular `SCollection`
       .toSCollection
+      // Save to the BigQuery table defined by "output" in the arguments passed in + "_team" suffix
       .saveAsTypedBigQuery(args("output") + "_team")
 
     gameEvents
+      // Recalculate using global window, 10 minutes after every first processing of an entry
```

Comment:
Regarding "first processing of an entry ..." please read [this](https://beam.apache.org/documentation/programming-guide/#processing-time-triggers), do you think your comment is clear in the sense given in the official doc?

Also don't hesitate to add links in to doc like the one above and:
 * https://beam.apache.org/documentation/programming-guide/#setting-a-trigger